source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2475 @@
+
+## Key Vault collection
+
+The key vault collection is the specially designated collection containing encrypted data keys. There is no default
+collection (user must specify). The key vault collection is used for automatic and explicit encryption/decryption as
+well as key management functions.
+
+For key management functions that require creating, updating, or deleting key documents in the key vault collection, the
+corresponding operations MUST be done with write concern majority.
+
+For encryption/decryption and key management functions that require reading key documents from the key vault collection,
+the corresponding operations MUST be done with read concern majority.
+
+Some key management functions may require multiple commands to complete their operation. Key management functions
+currently assume there is no concurrent access of the key vault collection being operated on. Concurrent access to a key
+vault collection being operated on may result in unexpected or undefined behavior. Support for concurrent key management
+may be considered for future work.
+
+See [Support sessions in key management functions](#support-sessions-in-key-management-functions).
+
+## Auto encrypt and decrypt
+
+An encrypted MongoClient automatically encrypts values for filtering and decrypts results.
+
+The driver MUST use libmongocrypt to initiate auto encryption and decryption. Create the BSON command meant to be sent
+over the wire, then pass that through the libmongocrypt state machine and use the returned BSON command in its place.
+The state machine is created with the libmongocrypt function `mongocrypt_ctx_new` and initialized with a
+`mongocrypt_ctx_encrypt_init` or `mongocrypt_ctx_decrypt_init`. See the
+[libmongocrypt API documentation](https://github.com/mongodb/libmongocrypt/blob/master/src/mongocrypt.h) for more
+information.
+
+An encrypted MongoClient configured with bypassAutoEncryption MUST NOT attempt automatic encryption for any command.
+
+Otherwise, an encrypted MongoClient MUST attempt to auto encrypt all commands. Note, the underlying implementation may
+determine no encryption is necessary, or bypass many checks if the command is deemed to not possibly contain any
+encrypted data (e.g. ping). See the appendix section:
+[libmongocrypt: Auto Encryption Allow-List](#libmongocrypt-auto-encryption-allow-list).
+
+An encrypted MongoClient MUST attempt to auto decrypt the results of all commands.
+
+Drivers MUST raise an error when attempting to auto encrypt a command if the maxWireVersion is less than 8. The error
+message MUST contain "Auto-encryption requires a minimum MongoDB version of 4.2".
+
+Note, all client side features (including all of `ClientEncryption`) are only supported against 4.2 or higher servers.
+However, errors are only raised for automatic encryption/decryption against older servers. See
+[Why is a 4.2 server required?](#why-is-a-4-2-server-required).
```

Comment:
```suggestion
[Why is a 4.2 server required?](#why-is-a-42-server-required).
```

Removing the `"pattern": "^#"` from `markdown_link_config.json` and running `pre-commit run --file ./source/client-side-encryption/client-side-encryption.md markdown-link-check` helped to identify dead relative links:
```
  ERROR: 11 dead links found!
  [✖] #why-is-a-4-2-server-required → Status: 404
  [✖] #extraoptions.cryptsharedlibrequired → Status: 404
  [✖] #ClientEncryptionOpts → Status: 404
  [✖] #AutoEncryptionOpts → Status: 404
  [✖] #AWSKMSOptions → Status: 404
  [✖] #GCPKMSOptions → Status: 404
  [✖] #AzureAccessToken → Status: 404
  [✖] #GetEncryptedFields → Status: 404
  [✖] #KMSProvider → Status: 404
  [✖] #masterkey2 → Status: 404
  [✖] #extraoptions.cryptsharedlibpath → Status: 404
```

Most seem to be false positives. I expect the check does not consider the `<div>` elements valid links.

Reply:
We now have an anchor checker, from #1531, I'm going to merge from master and address the failures.